formula-android/src/main/java/com/instacart/formula/android/ViewInstance.kt

```diff
@@ -0,0 +1,59 @@
+package com.instacart.formula.android
+
+import android.view.View
+import com.instacart.formula.RenderView
+import com.instacart.formula.Renderer
+import com.instacart.formula.android.views.FeatureViewBindFunction
+import com.instacart.formula.fragment.FormulaFragment
+import com.instacart.formula.fragment.FragmentLifecycleCallback
+
+/**
+ * View instance contains an initialized Android [view] and provides factory
+ * [methods][featureView] to create a [FeatureView].
+ */
+abstract class ViewInstance {
+    /**
+     * Android view which will be used to create [FeatureView]. This view will be returned
+     * as part of [FormulaFragment.onCreateView].
+     */
+    abstract val view: View
+
+    /**
+     * Creates a [FeatureView] from a [render] function
+     */
+    fun <RenderModel> featureView(
+        lifecycleCallbacks: FragmentLifecycleCallback? = null,
+        render: (RenderModel) -> Unit
```

Comment:
Worth swapping params order here to align with other `featureView()` counterparts.

Reply:
Having lambda as last param here allows better Kotlin structure such as 
```kotlin
featureView(lifecycleCallbacks) { renderModel ->
  // render
}
```

Reply:
Might be worth updating the others to ask for lifecycle callbacks first, but it would make the other API usage not ideal. For example, if I reverse it to 
```kotlin
featureView(
  lifecycleCallbacks: FragmentLifecycleCallbacks? = null,
  renderView: RenderView
)
```
I likely won't be able to call it `featureView(MyRenderView())`